src/arizona_live_reload.erl

```diff
@@ -39,57 +37,69 @@ Live-reload functionality for use during development.
 -export([handle_info/2]).
 
 %% State
--record(state, {timer, files = #{}, clients = #{}}).
-
-%% Macros
--define(SERVER, ?MODULE).
+-record(state, {
+    timer :: undefined | reference(),
+    files :: #{string() := {erl, modified}},
+    clients :: #{pid() := boolean()}
+}).
+-opaque state() :: #state{}.
+-export_type([state/0]).
+-elvis([{elvis_style, state_record_and_type, disable}]). % opaque not identified as "type"
 
 %% --------------------------------------------------------------------
 %% API functions.
 %% --------------------------------------------------------------------
 
+-spec start_link() -> gen_server:start_ret().
 start_link() ->
-    gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
```

Comment:
Just so that I don't have to open an Elvis exception. Lemme know if you feel a variable named `Server` is more explicit.

Reply:
I don't have a strong opinion on this, but it sounds more natural to me to call it `Server`, for example:
```erlang
gen_server:cast(?SERVER, reload).
% vs
gen_server:cast(?MODULE, reload).
```
But it does not make too much of a difference to me. Feel free to decide on this :slightly_smiling_face: 
The supervisor also does not use a `?SUPERVISOR` macro.

Reply:
Otoh, `?MODULE` is more portable, if you wanna move your code, since you don't have to move the macro with it. I'll keep `SERVER` since it's meaningful to you.

Reply:
9005691